test/models/user_test.rb

```diff
@@ -762,6 +762,23 @@ def setup_user(operation, type = 'users', search = nil)
         end
       end
 
+      test 'existing user of different authsource' do
+        user = User.new(:login => "dummy", :mail => "j@j.com",
+                        :auth_source_id => AuthSourceInternal.first.id,
+                        :organizations => User.current.organizations,
+                        :locations => User.current.locations)
+        user.password_hash = "asd"
+        user.save!
```

Comment:
Why not use `FactoryBot` to create the user for you?

Reply:
sure.